packages/datasource-sparql/lib/datasources/SparqlDatasource.js

```diff
@@ -49,10 +49,10 @@ class SparqlDatasource extends Datasource {
         response.results.bindings.forEach(function (binding) {
           binding = self._sparqlJsonParser.parseJsonBindings(binding);
           var triple = {
-            subject:   binding.s ? termToString(binding.s) : query.subject,
-            predicate: binding.p ? termToString(binding.p) : query.predicate,
-            object:    binding.o ? termToString(binding.o) : query.object,
-            graph:     binding.g ? termToString(binding.g) : query.graph,
+            subject:   binding.s ? binding.s : query.subject,
```

Comment:
Patterns like this can be simplified as follows:
```suggestion
            subject:   binding.s || query.subject,
```